Rubberduck.RegexAssistant/IRegularExpression.cs

```diff
@@ -0,0 +1,327 @@
+﻿using Rubberduck.RegexAssistant.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Rubberduck.RegexAssistant
+{
+    public interface IRegularExpression : IDescribable
+    {
+        Quantifier Quantifier { get; }
+
+        bool TryMatch(string text, out string remaining);
+    }
+
+    public class ConcatenatedExpression : IRegularExpression
+    {
+        private readonly Quantifier quant;
+        internal readonly IList<IRegularExpression> Subexpressions;
+
+        public ConcatenatedExpression(IList<IRegularExpression> subexpressions)
+        {
+            Subexpressions = subexpressions;
+            quant = new Quantifier(""); // these are always exactly once. Quantifying happens through groups
+        }
+
+        public string Description
+        {
+            get
+            {
+                return string.Join(", ", Subexpressions.Select(exp => exp.Description)) + " " + Quantifier.HumanReadable();
+            }
+        }
+
+        public Quantifier Quantifier
+        {
+            get
+            {
+                return quant;
+            }
+        }
+
+        public bool TryMatch(string text, out string remaining)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class AlternativesExpression : IRegularExpression
+    {
+        private readonly Quantifier quant;
+        internal readonly IList<IRegularExpression> Subexpressions;
+
+        public AlternativesExpression(IList<IRegularExpression> subexpressions)
+        {
+            Subexpressions = subexpressions;
+            quant = new Quantifier(""); // these are always exactly once. Quantifying happens through groups
```

Comment:
actually not a nitpick. using "" impacts performance on lower optimization compilations because it actually allocates a new String every time